Tests/CatchToResultTests.swift

```diff
@@ -0,0 +1,153 @@
+//
+//  CatchToResult.swift
+//  CombineExt
+//
+//  Created by Yurii Zadoianchuk on 05/03/2021.
+//  Copyright © 2021 Combine Community. All rights reserved.
+//
+
+import Foundation
+
+#if !os(watchOS)
+import XCTest
+import Combine
+import CombineExt
+
+@available(OSX 10.15, iOS 13.0, tvOS 13.0, watchOS 6.0, *)
+final class CatchToResultTests: XCTestCase {
+    private var subscription: AnyCancellable!
+
+    enum CatchToResultError: Error {
+        case someError
+    }
+
+    func testCatchToResult() {
+        let subject = PassthroughSubject<Int, Error>()
+        let testInt = 5
+        var completed = false
+        var results: [Result<Int, Error>] = []
+
+        subscription = subject
+            .catchToResult()
+            .sink(receiveCompletion: { _ in completed = true },
+                  receiveValue: { results.append($0) })
+
+        subject.send(testInt)
+        XCTAssertFalse(completed)
+        subject.send(testInt)
+        subject.send(completion: .finished)
+        XCTAssertTrue(completed)
+        XCTAssertEqual(results.count, 2)
+        let intsCorrect = results.compactMap({ try? $0.get() }).allSatisfy({ $0 == testInt })
+        XCTAssertTrue(intsCorrect)
+    }
+
+    func testCatchCustomError() {
+        let subject = PassthroughSubject<Int, Error>()
+        var completed = false
+        var gotFailure = false
+        var gotSuccess = false
+        var result: Result<Int, Error>? = nil
+
+        subscription = subject
+            .tryMap({ _ -> Int in throw(CatchToResultError.someError) })
```

Comment:
```suggestion
            .tryMap { _ -> Int in throw CatchToResultError.someError }
```

please drop parentheses in trailing-closures everywhere

Also, throw isn't a function, it's a specialty word.